tests/Doctrine/Tests/DBAL/Functional/Driver/Mysqli/MysqliStatementTest.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Functional\Driver\Mysqli;
+
+use Doctrine\DBAL\Driver\Mysqli\Driver;
+use Doctrine\DBAL\FetchMode;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\Tests\DbalFunctionalTestCase;
+use Throwable;
+use const CASE_LOWER;
+use function array_change_key_case;
+use function extension_loaded;
+
+final class MysqliStatementTest extends DbalFunctionalTestCase
+{
+    /**
+     * @return iterable<string, array<string, mixed[], mixed[]>>
+     */
+    public static function prepareProvider() : iterable
+    {
+        return [
+            'single named parameter' => [
+                'SELECT * FROM mysqli_statement_test f WHERE f.foo = :foo',
+                ['foo' => 1],
+                [
+                    ['id' => 1, 'foo' => 1, 'bar' => 1],
+                    ['id' => 2, 'foo' => 1, 'bar' => 2],
+                    ['id' => 3, 'foo' => 1, 'bar' => 3],
+                    ['id' => 4, 'foo' => 1, 'bar' => 4],
+                ],
+            ],
+
+            'multiple parameters' => [
+                'SELECT * FROM mysqli_statement_test f WHERE f.foo = :foo AND f.bar = :bar',
+                [
+                    'foo' => 1,
+                    'bar' => 2,
+                ],
+                [
+                    ['id' => 2, 'foo' => 1, 'bar' => 2],
+                ],
+            ],
+
+            'same parameter at multiple positions' => [
+                'SELECT * FROM mysqli_statement_test f WHERE f.foo = :foo AND f.foo IN (:foo)',
+                ['foo' => 1],
+                [
+                    ['id' => 1, 'foo' => 1, 'bar' => 1],
+                    ['id' => 2, 'foo' => 1, 'bar' => 2],
+                    ['id' => 3, 'foo' => 1, 'bar' => 3],
+                    ['id' => 4, 'foo' => 1, 'bar' => 4],
+                ],
+            ],
+
+            'parameter with string value' => [
+                'SELECT * FROM mysqli_statement_test f WHERE f.foo = :foo',
+                ['foo' => '"\''],
+                [],
+            ],
+        ];
```

Comment:
What happens if there are more parameters than needed in the query? Or when there are fewer than needed (an exception I hope).

Reply:
Too few parameters:

```
Doctrine\DBAL\DBALException: An exception occurred while executing 'SELECT * FROM mysqli_statement_test f WHERE f.foo = :foo':

Wrong parameter count for mysqli_stmt::bind_param()
```

Too many parameters:

```
Doctrine\DBAL\DBALException: An exception occurred while executing 'SELECT * FROM mysqli_statement_test f WHERE f.foo = :foo' with params [1, 2]:

Undefined index: bar
```

where `bar` is the name of the superfluous named parameter.

Are these exceptions okay? Do you want a test for them?

Reply:
I will answer once and if https://github.com/doctrine/dbal/pull/3891#issuecomment-606299763 is resolved